### PR TITLE
ref(replays): refactor replay onboarding docs to be separate

### PR DESCRIPTION
**Pull request by michellewzhang**
The replay onboarding content is now fairly separate from the normal onboarding docs -- the only big thing we're reusing is the `Step` component.

- Gets rid of `SdkDocumentation` usage in our sidebar file
- Instead, use a new hook `useLoadOnboardingDoc` to load the new onboarding content we want
- No behavior change (I hope)
- Create our own replay-specific `ReplayOnboardingLayout` component instead of re-using the docs one
- The boolean `isReplayConfigStep` is gone; we don't need it anymore
- Moved the 'Mask All Text' and 'Block All Media' toggles to right above the code snippet

https://github.com/getsentry/sentry/assets/56095982/fb1a2d80-6d15-440e-97c6-da95c226d29d

### CODE REVIEW

File: static/app/components/onboarding/gettingStartedDoc/replayConfigToggle.tsx
Comment (ArthurKnaus) [inline static/app/components/onboarding/gettingStartedDoc/replayConfigToggle.tsx:1]: You can move this file to the `replaysOnboarding` folder

File: static/app/components/replaysOnboarding/useLoadOnboardingDoc.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        : platform?.id.replace(`${platform.language}-`, `${platform.language}/`)
+      : `${platform?.language}/${platform?.id}`;
+
+  const {
+    data: projectKeys,
+    isError: projectKeysIsError,
+    isLoading: projectKeysIsLoading,
+  } = useApiQuery<ProjectKey[]>([`/projects/${organization.slug}/${projectSlug}/keys/`], {
+    staleTime: Infinity,
+  });
```
Comment (ArthurKnaus) [inline static/app/components/replaysOnboarding/useLoadOnboardingDoc.tsx:33]: Nit: I have the feeling this should not be part of this hook (separation of concerns) 🤔

**Review by ArthurKnaus — APPROVED** [review]
Huge thx for refactoring this! 🥇

File: static/app/components/replaysOnboarding/useLoadOnboardingDoc.tsx (hunk 1 above)
Comment (michellewzhang) [reply static/app/components/replaysOnboarding/useLoadOnboardingDoc.tsx:33]: yeah this API call could definitely be extracted into a hook, it's used in a lot of places!

**Pull request merged**